internal/controller/ingress/httpsedge_controller.go

```diff
@@ -1079,11 +1092,30 @@ func (u *edgeRouteModuleUpdater) setEdgeRoutePolicy(ctx context.Context, route *
 		return client.Delete(ctx, edgeRouteItem(route))
 	}
 
-	log.Info("Updating Policy module")
-	_, err := client.Replace(ctx, &ngrokapi.EdgeRoutePolicyRawReplace{
+	module := ngrok.EndpointTrafficPolicy{}
+
+	if util.IsLegacyPolicy(trafficPolicy) {
+		u.recorder.Eventf(u.edge, v1.EventTypeWarning, "UpdateWarning", "Traffic Policy is using legacy directions: ['inbound', 'outbound']. Update to new phases: ['on_tcp_connect', 'on_http_request', 'on_http_response']")
+	}
+
+	// legacy version of the controller store "enabled" in the payload, which needs to be extracted
+	trafficPolicy, enabledFieldVal, err := util.ExtractEnabledField(trafficPolicy)
+	if err != nil {
+		return err
+	}
+	module.Enabled = enabledFieldVal
+
+	if enabledFieldVal != nil {
+		u.recorder.Eventf(u.edge, v1.EventTypeWarning, "UpdateWarning", "Traffic Policy has 'enabled' set. This is a legacy option that will stop being supported soon.")
+	}
+
+	module.Value = string(trafficPolicy)
+
+	log.Info("Updating Traffic Policy module")
```

Comment:
Nit: Can we add a k8s event here for success on the object too?

Reply:
yep, i'll change this!